### ISSUE TITLE
[TF 2.0.0rc0] Cannot connect to TPU device

### ISSUE THREAD

**Issue by nsantavas** [tensorflow/tensorflow#32043]
Created VM and v3-8 TPU with ctpu up command and updated TF version to TF2.0.0rc0 via pip3. 
When i try to connect to tpu device returns error:
```
InvalidArgumentError: Unable to find a context_id matching the specified one (5613663074031560004). Perhaps the worker was restarted, or the context was GC'd?
Additional GRPC error information:
{"created":"@1566994715.938381293","description":"Error received from peer","file":"external/grpc/src/core/lib/surface/call.cc","file_line":1039,"grpc_message":"Unable to find a context_id matching the specified one (5613663074031560004). Perhaps the worker was restarted, or the context was GC'd?","grpc_status":3}
2019-08-28 12:18:36.196440: E tensorflow/core/distributed_runtime/rpc/eager/grpc_eager_client.cc:72] Remote EagerContext with id 5613663074031560004 does not seem to exist.
```
I also tried the same in Colab, with rc0 version and i have the same error. The code i used is the one given in documentation:

```
tpu='test'
resolver = tf.distribute.cluster_resolver.TPUClusterResolver(tpu=tpu)
tf.config.experimental_connect_to_host(resolver.master())
tf.tpu.experimental.initialize_tpu_system(resolver)
tpu_strategy = tf.distribute.experimental.TPUStrategy(resolver)
```

**Comment by jmgc** [comment]
Same situation here. I am trying to train a NN in a TPU in eager mode. Following the advice given in #30129 by @capilano, I switched to the nightly build: 2.0.0-dev20190731.

**Comment by rishabhsahrawat** [comment]
@jmgc TPUs do not support eager execution. Try training without enabling it. For more info look at the question 11 [here](https://cloud.google.com/tpu/docs/faq).
@nsantavas maybe you are also using eager execution?

**Comment by nsantavas** [comment]
@rishabhsahrawat no

**Comment by capilano** [comment]
If you are using TF2.0 , tf.function will automatically handle graphs, so the input pipeline can be executed in eager mode.

**Comment by pshiko** [comment]
I am currently facing the same problem.
I also use v3-8 TPU and tf2.0 rc0.

@capilano 

Which version of tensorflow can run [that link code](https://github.com/tensorflow/tpu/blob/master/models/experimental/resnet50_keras/resnet50_tf2.py)?
I think this is not tf2.0 code because `import tensorflow.compat.v2 as tf` and ` tf.enable_v2_behavior()` are called in the source code.
https://github.com/tensorflow/tpu/blob/aada81f730478e66ed736aea5db33fc9895341a4/models/experimental/resnet50_keras/resnet50_tf2.py#L34
https://github.com/tensorflow/tpu/blob/aada81f730478e66ed736aea5db33fc9895341a4/models/experimental/resnet50_keras/resnet50_tf2.py#L170

@goldiegadde 
I want to know the status of TPU support on tensorflow 2.0rc0.
I have no other information than release notes and tensorflow guide.

https://github.com/tensorflow/tensorflow/releases
https://www.tensorflow.org/beta/guide/distribute_strategy

In the current situation I don't know if this is a bug to report or just not supported.

**Comment by capilano** [comment]
@pshiko I am not really sure, I think you can call that in 2.0 and 1.0. I have seen TF2.0 and TPU issues, so there is support, maybe there is a bug. Or possibly you could try TF 2.0-nightly.
Personally, I do not have access to  a TPU that has tensorflow 2.0, so all my code is on 1.14 and I have eager disabled. Apparently later version of 1.0 also support eager training similar to 2.0 if you call those lines.

**Comment by frankchn** [comment]
Hello! I don't think there is an official TPU release for 2.0 (i.e. you cannot specify --version=2.0 when creating a TPU with ctpu or gcloud) so it is currently unsupported.

**Comment by frankchn** [comment]
In general, we anticipate that TF 2.1 (coming later this year) will have much better support for TPUs than the initial 2.0 release.

**Comment by nsantavas** [comment]
Yes, but if you update VM host via pip to TF-nightly-2.0.0rc0 ?

**Comment by frankchn** [comment]
No, the TPU itself must have the same version as the VM host, or the configuration is not supported.

**Comment by pshiko** [comment]
@frankchn 
Thank you for your kind reply !!
I understood the current status of TPU support. 

According to https://www.tensorflow.org/beta/guide/distribute_strategy , `TPUStrategy` with `Keras API` support is planned in 2.0RC.
But the situation seems to have change.

I will consider using TF1.x for a while.
Thank you!!

**Comment by frankchn** [comment]
@pshiko Yeah, we have been working on that, but only a small subset of customers have access to the internal TF2.0 image for the TPU itself, so it is not broadly available yet.

Otherwise, running a TF2.0 VM host with a TF1.14 TPU is not compatible because the 1.14 TPU (having been released months prior to any 2.0 release) would lack code that supports the new features in 2.0.
